### DISCUSSION TITLE
Refactor `any(item for item in items if item in container)` into `any(item in container for item in items)`

### DISCUSSION THREAD

**Discussion by brettcannon** [astral-sh/ruff#7998] [Ideas]
Came up in a code review for `any(p for p in parts if p in s)`. With no assignment expression or any need to know which or how many items match since the loop variant doesn't leak out of the generator expression, there's no need to condition the `any()` call on succeeding via which items matched.

**Comment by JonathanPlasse** [comment]
Hi, this is not equivalent if some items are falsy.
```console
>>> items = [True, False]
>>> container = [False]
>>> any(item for item in items if item in container)
False
>>> any(item in container for item in items)
True
```

**Reply by brettcannon** [reply]
True, although I think you're asking for trouble in that situation. But I see why Ruff  wouldn't want to risk the pathological case of being wrong here.
